flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/crd/status/FlinkDeploymentStatus.java

```diff
@@ -26,6 +28,8 @@
 @NoArgsConstructor
 @AllArgsConstructor
 public class FlinkDeploymentStatus {
-    private JobStatus jobStatus;
+    private JobStatus jobStatus = new JobStatus();
```

Comment:
Not a crucial one: having default value for `jobStatus` field or not both make sense to me. But I am a little curious if there is any specific reason for adding it here. If there is, maybe we can refactor `createJobStatus()` in Observer.java to a static method here like `parseFromJobMessage()`. Whatever, this one is not critical for this PR and we can leave it as a followup if we hope.

Reply:
Having a default a jobstatus avoids a bunch of null checks, and also not that you mention it I simplified the status update in a commit: https://github.com/apache/flink-kubernetes-operator/pull/26/commits/d4924c0849976930d591a6f60ab77759a4582e5b